png/build.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+set -e
+
+
+extract_sources() {
+  if [ ! -d "${PREFIX_PORT_SRC}" ]; then
+    echo "Extracting sources from ${archive_filename}"
```

Comment:
nitpick: Just wondering whether this function is generic enough to make it accept an argument (a filename to extract) instead of using a "hoisted" global variable. Maybe this will increase readability in case someone glances from top to bottom.

Reply:
By "a filename to extract" I meant `${archive_filename}`. I assumed `${PREFIX_PORT_SRC}` can be used as a global, as is the case in the other ports' build.sh scripts.

Usually, ports have a simple build pipeline which doesn't have to be separated into functions etc. - in this case globals are defined and used directly, but I understand that some ports (tinyx - I see you) need more encapsulation.